usaspending_api/etl/management/commands/reindex_elastic.py

```diff
@@ -0,0 +1,157 @@
+from pprint import pprint
+
+logging.basicConfig(format='%(asctime)s |  %(message)s', level=logging.WARNING)
+MAX_RETRIES = 3
+INGEST_RATE = 10000
+
+ES_CLIENT = Elasticsearch(settings.ES_HOSTNAME, timeout=300)
+INDICES_SUB_NAMES = indices_to_award_types.keys()
+
+
+class Command(BaseCommand):
+    # used by parent class
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dest_prefix',
+            type=str,
+            help='Prefix of output index')
+        parser.add_argument(
+            '--source_prefix',
+            default=settings.TRANSACTIONS_INDEX_ROOT,
+            type=str,
+            help='Prefix of index that has data')
+        parser.add_argument(
+            '--mapping_file',
+            type=str,
+            help='Absolute path to new mapping file')
+        parser.add_argument(
+            '--alias_prefix',
+            default='trans',
+            type=str,
+            help='Mapping file')
+
+    def handle(self, *args, **options):
+        ''' Script execution of custom code starts in this method'''
+        set_config()
+        self.source_prefix = options['source_prefix']
+        self.dest_prefix = options['dest_prefix']
+        self.alias_prefix = options['alias_prefix']
+        # TODO:
+        self.wipe = False
+        with open(options['mapping_file']) as f:
+            data = json.load(f)
+            self.mapping_file = json.dumps(data)
+
+        self.indicies_dict()
+        self.controller()
+
+    def single_reindex(self, source, dest):
+        logging.warn('calling create')
+        ES_CLIENT.indices.create(dest, body=self.mapping_file)
+        body_ = {
+                "source": {
+                        "index": source
+                        },
+                "dest": {
+                        "index": dest
+                        }
+                }
+        logging.warn('CREATING INDEX {}'.format(dest))
+        ES_CLIENT.reindex(body=body_)
+
+    def indicies_dict(self):
+        index_name = self.source_prefix + '*'
+        source_indices = ES_CLIENT.indices.get(index_name).keys()
+        source_indices = sorted(source_indices, key=lambda x: x.split('-')[-1])
+
+        dest_indices = [i.replace(self.source_prefix, self.dest_prefix) for i in source_indices]
+        self.reindex_dict = dict(zip(source_indices, dest_indices))
+        count = self.get_total_count(index_name)
+        logging.warn('Reindexing a total of {} documents'.format(count))
+
+    def get_total_count(self, index_name):
+        try:
+            response = ES_CLIENT.search(index=index_name)
+            return response["hits"]["total"]
+        except TransportError as e:
+            logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                          format(e=str(e)))
+            return -1
+
+    def verify(self, source, target):
+        difference = self.get_total_count(source) - self.get_total_count(target)
```

Comment:
What will be the behavior if both of the get_total_count functions have errors and return -1?

Reply:
If it's a transport error then they'd both be -1..wouldn't sleep and pass to the next index to reindex.